src/core/src/MatrixDynSize.cpp

```diff
@@ -76,37 +77,32 @@ MatrixDynSize& MatrixDynSize::operator=(const MatrixDynSize& other)
 {
     if (this == &other) return *this;
 
-    // Copy the new rows
+    // Copy the new rows and columns
     m_rows = other.m_rows;
     m_cols = other.m_cols;
+    unsigned requiredCapacity = m_rows * m_cols;
+
+    // if other is empty, return
+    //FIXME: This probably changed the behaviour.
+    // the capacity is never changed if the other matrix is zero
+    if (requiredCapacity == 0) return  *this;
```

Comment:
Honestly I thought the contrary. It is usually better to do "early return" as it simplify code reading.

About the previous behaviour. I think that before, if `other` is  empty, then the internal buffer is freed, while now the capacity is not changed (which is usually a desired feature)